src/twister2/plugin.py

```diff
@@ -156,18 +172,22 @@ def pytest_configure(config: pytest.Config):
     # Export zephyr_base variable so other tools like west would also use the same one
     os.environ['ZEPHYR_BASE'] = zephyr_base
 
-    is_worker_input = hasattr(config, 'workerinput')  # xdist worker
+    worker_input = hasattr(config, 'workerinput')  # xdist worker
+
+    if not config.option.collectonly or worker_input:
+        choice = 'delete' if config.option.build_only else config.option.clear
```

Comment:
What in situation when user would like to apply `--build-only` and still keep historical artifacts/builds by `--clear=archive`?
Maybe should be:
```python
clear_choice = config.option.clear
if config.option.build_only and clear_choice == 'no':
    msg = 'To apply "--build-only" option, "--clear" option cannot be set as "no".'
    logger.error(msg)
    raise TwisterException(msg)
else:
    run_artifactory_cleanup(clear_choice, config.option.output_dir)
```